htdocs/admin/content/alliance/edit.inc.php

```diff
@@ -1,638 +1,534 @@
-  						<input type=\"radio\" name=\"output\" id=\"output\" value=\"0\" checked=\"checked\"/> Einzeln / <input type=\"radio\" name=\"output\" id=\"output\" value=\"1\"/> Summiert
-  					</td>
-  				</tr><tr>
-  					<th>Einzahlungen:</th>
-  					<td>
-		  				<select id=\"limit\" name=\"limit\">
-							<option value=\"0\" checked=\"checked\">alle</option>
-							<option value=\"1\">die letzte</option>
-							<option value=\"5\">die letzten 5</option>
-							<option value=\"20\">die letzten 20</option>
-						</select>
-					</td>
-				</tr><tr>
-  					<th>Von User:</th>
-  					<td>
-  						<select id=\"user_spends\" name=\"user_spends\">
-							<option value=\"0\">alle</option>";
-					  	// Allianzuser
-							foreach($members as $mid => $data)
-							{
-					  		echo "<option value=\"".$mid."\">".$data['user_nick']."</option>";
-					  	}
-  			echo 		"</select>
-  					</td>
-	  			</tr><tr>";
-			 tableEnd();
-			 echo "<p><input type=\"button\" onclick=\"xajax_showSpend(".$arr['alliance_id'].",xajax.getFormValues('filterForm'))\" value=\"Anzeigen\"\"/></p>";
-			 echo "</form>";
-
-			 echo "<div id=\"spends\">&nbsp;</div>";
-
-			echo '</div><div id="tabs-7">';
-
-			/**
-			* Gebäude
-			*/
-			$res = dbquery("
-						SELECT
-							alliance_buildlist.*,
-							alliance_buildings.alliance_building_name
-						FROM
-							alliance_buildlist
-						INNER JOIN
-							alliance_buildings
-						ON
-							alliance_buildings.alliance_building_id=alliance_buildlist.alliance_buildlist_building_id
-							AND	alliance_buildlist_alliance_id='".$id."';");
-
-			$buildings = dbquery("
-						SELECT
-							alliance_building_id,
-							alliance_building_name
-						FROM
-							alliance_buildings;");
-
-			tableStart();
-			echo "<tr>
-					<th>Gebäude</th><th>Stufe</th><th>Useranzahl</th><th>Status</th>
-				</tr>";
-			if (mysql_num_rows($res)>0)
-			{
-				while ($arr = mysql_fetch_assoc($res))
-				{
-					echo "<tr><td>".$arr['alliance_building_name']."</td><td>".$arr['alliance_buildlist_current_level']."</td><td>".$arr['alliance_buildlist_member_for']."</td><td>";
-					if ($arr['alliance_buildlist_build_end_time']>time()) echo "Bauen";
-					elseif ($arr['alliance_buildlist_build_end_time']>0) echo "Bau abgeschlossen";
-					else echo "Untätig";
-					echo "</td>";
-					echo "</tr>";
-
-				}
-			}
-			else
-				echo "<tr><td colspan=\"4\">Keine Gebäude vorhanden!</td></tr>";
-
-			tableEnd();
+	echo '<br><input type="submit" name="techs">';
 
-		    echo '<br><h2>Gebäude hinzufügen</h2>';
+	echo '
+		</div>
+	</div>';
+}
 
-			tableStart();
-
-            echo "<tr>
-					<th>Gebäude</th><th>Stufe</th><th>Useranzahl</th>
-				</tr>";
-			echo'<tr><td>';
-
-            if (mysql_num_rows($buildings)>0)
-			{
-				echo'<select name="selected">';
-				while ($arr = mysql_fetch_assoc($buildings))
-				{
-					echo "<option>".$arr['alliance_building_name']."</option>";
+function infoTab(array $arr, array $members): void
```

Comment:
Instead of `$arr` can we use more specific names?

Reply:
Done 1d376f5